Sources/DatadogSDKTesting/IntelligentTestRunner/GitUploader.swift

```diff
@@ -96,82 +119,87 @@ struct GitUploader {
         log.debug("Git status: \(status)")
         return status.isEmpty
     }
-
-    private func handleShallowClone(repository: String) -> Bool {
+    
+    private func searchForNewCommits(repositoryURL: String) -> [String]? {
+        guard let latestCommits = log.measure(name: "getLatestCommits", getLatestCommits),
+              !latestCommits.isEmpty else
+        {
+            log.print("sendGitInfo failed, can't get latest commits")
+            return nil
+        }
+        let existingCommits = log.measure(name: "searchRepositoryCommits") {
+            exporter.searchCommits(repositoryURL: repositoryURL, commits: latestCommits)
+        }
+        let commits = Set(latestCommits).subtracting(existingCommits)
+        if commits.isEmpty { return [] }
+        return log.measure(name: "getCommitsAndTrees") {
+            getCommitsAndTrees(included: Array(commits), excluded: existingCommits)
+        }
+    }
+    
+    private var isShallowRepository: Bool {
         // Check if is a shallow repository
-        guard let isShallow = Spawn.output(
-            try: #"git -C "\#(workspacePath)" rev-parse --is-shallow-repository"#, log: log
-        ) else {
+        guard let isShallow = git("rev-parse --is-shallow-repository") else {
             return false
         }
         log.debug("isShallow: \(isShallow)")
-        guard isShallow == "true" else {
-            return true
-        }
-
+        return isShallow == "true"
+    }
+    
+    private func handleShallowClone(repository: String) -> Bool {
         // Count if number of returned lines is greater than 1
-        guard let lineLength = Spawn.output(
-            try: #"git -C "\#(workspacePath)" log --format=oneline -n 2"#, log: log
-        ) else {
+        guard let last2Commits = git("log --format=oneline -n 2") else {
             return false
         }
-        log.debug("lineLength: \(lineLength)")
-        if lineLength.contains("\n") {
+        log.debug("last2Commits: \(last2Commits)")
+        if last2Commits.contains("\n") {
```

Comment:
I am wondering about this check: sometimes we have shallow clones that contain more than 2 commits (for example web-ui shallow clones have 5 commits by default). This optimization might not be what you want as shallow clone could contain more than 1 commit.

Reply:
yeah. It's an old code and I think you are right. It's a bad optimisation. Removed